docs/reference/commands/keystore.asciidoc

```diff
@@ -0,0 +1,208 @@
+
+[discrete]
+[[elasticsearch-keystore-examples]]
+=== Examples
+
+[discrete]
+[[creating-keystore]]
+==== Create the keystore
+
+To create the `elasticsearch.keystore`, use the `create` command:
+
+[source,sh]
+----------------------------------------------------------------
+bin/elasticsearch-keystore create -p
+----------------------------------------------------------------
+
+You are prompted to enter the keystore password. A password-protected
+`elasticsearch.keystore` file is created alongside the `elasticsearch.yml` file.
+
+[discrete]
+[[changing-keystore-password]]
+==== Change the password of the keystore
+
+To change the password of the `elasticsearch.keystore`, use the `passwd` command:
+
+[source,sh]
+----------------------------------------------------------------
+bin/elasticsearch-keystore passwd
+----------------------------------------------------------------
+
+If the {es} keystore is password protected, you are prompted to enter the
+current password and then enter the new one. If it is not password protected,
+you are prompted to set a password.
+
+[discrete]
+[[list-settings]]
+==== List settings in the keystore
+
+To list the settings in the keystore, use the `list` command.
+
+[source,sh]
+----------------------------------------------------------------
+bin/elasticsearch-keystore list
+----------------------------------------------------------------
+
+If the {es} keystore is password protected, you are prompted to enter the
+password.
+
+[discrete]
+[[add-string-to-keystore]]
+==== Add settings to the keystore
+
+Sensitive string settings, like authentication credentials for Cloud plugins,
+can be added with the `add` command:
+
+[source,sh]
+----------------------------------------------------------------
+bin/elasticsearch-keystore add the.setting.name.to.set
+----------------------------------------------------------------
+
+You are prompted to enter the value of the setting. If the {es} keystore is
+password protected, you are also prompted to enter the password.
+
+To pass the setting value through standard input (stdin), use the `--stdin` flag:
+
+[source,sh]
+----------------------------------------------------------------
+cat /file/containing/setting/value | bin/elasticsearch-keystore add --stdin the.setting.name.to.set
+----------------------------------------------------------------
+
+You can overwrite existing entries in the keystore by using the `-f` parameter:
+
+[source,sh]
+----------------------------------------------------------------
+bin/elasticsearch-keystore add -f the.existing.setting.name.to.set
+----------------------------------------------------------------
+
+NOTE: The `-f` parameter will also force the creation of an obfuscated-only
+keystore, if one doesn't already exist.
+
+////
+TBD: This NOTE is unclear.  Why would you want both a password-protected and an
+obfuscated-only keystore? Or does this mean that it creates the keystore if it
+doesn't already exist?
```

Comment:
It's the latter. Let's say you don't have a keystore file.

1. If you run this command without the `-f` flag, you get a prompt asking whether you want to create a keystore. Type `N` and the command exits, or type `y` and proceed.
2. With the `-f` flag, the command creates a non-password-protected ("obfuscated-only") keystore in the background.

If you already have a keystore, this command won't change its password-protection status.

Reply:
Thanks for the clarification!  I've updated the text